cds_books/ldap/api.py

```diff
@@ -175,24 +190,183 @@ def _commit_user(user_data):
-                "CERN_APP_CREDENTIALS", {}).get("consumer_key") or "CLIENT_ID"
+            client_id = (
+                current_app.config.get("CERN_APP_CREDENTIALS", {}).get(
+                    "consumer_key"
+                ) or "CLIENT_ID"
+            )
             remote_account = RemoteAccount(
-                                client_id=client_id,
-                                **self.import_remote_account(
-                                    user_id, ldap_user))
+                client_id=client_id,
+                **self.import_remote_account(user_id, ldap_user)
+            )
             db.session.add(remote_account)
 
         db.session.commit()
+
+
+def index_ldap_users():
+    """Index ldap users in ES."""
+    from invenio_base.app import create_cli
+
+    cli = create_cli()
+    runner = current_app.test_cli_runner()
+    command = "ils patrons index"
+    click.secho("ils {}...".format(command), fg="green")
+    runner.invoke(cli, command, catch_exceptions=True)
+    _log_info("command_executed", dict(command=command))
+
+
+def import_ldap_users(ldap_users):
+    """Import ldap users in db."""
+    importer = LdapUserImporter(ldap_users)
+    importer.import_users()
+    click.secho("Now indexing...", fg="green")
+    index_ldap_users()
+    _log_info("users_indexed")
+
+
+def check_user_for_update(system_user, ldap_user):
+    """Check if there is an ldap update for a user and commit changes."""
+    ldap_user_department = ldap_user["department"][0].decode("utf8")
+    if not system_user.extra_data["department"] == ldap_user_department:
+        previous_department = system_user.extra_data["department"]
+
+        click.secho(
+            "Changes detected for system user {}".format(
+                str(system_user.user)
+            ),
+            fg="green",
+        )
+        click.secho(
+            "System user's department {} is different than the {}".format(
+                previous_department, ldap_user_department
+            ),
+            fg="green",
+        )
+
+        system_user.extra_data.update(dict(department=ldap_user_department))
+        db.session.commit()
+
+        _log_info("department_updated",
+                  dict(
+                      user_id=system_user.user.id,
+                      previous_department=previous_department,
+                      new_department=ldap_user_department
+                  ))
+
+
+def delete_user(system_user):
+    """Delete a system user."""
+    pass
+
+
+def _log_info(action, extra=None):
+    if extra is None:
+        extra = dict()
+    name = "ldap_users_synchronization"
+    structured_msg = dict(
+        name=name,
+        uuid=str(uuid.uuid4()),
+        action=action,
+        **extra,
+    )
+    structured_msg_str = json.dumps(structured_msg, sort_keys=True)
+    current_app.logger.info(structured_msg_str)
+
+
+def _log_exception(exception):
+    current_app.logger.exception(exception)
+
+
+def sync_users():
```

Comment:
Same thought as above about if this should be defined here or in `cli.py`! Maybe also @kprzerwa @KonstantinaStoikou @jrcastro2 can share their thoughts!

Reply:
cli would be more intuitive for us I think